File: 01. Binary search/main.py
def search(arr, item_to_find):
    low = 0
    high = len(array) - 1

    while low < high:
        mid = int((low + high) / 2)
        guess = arr[mid]
        if guess == item_to_find:
            return mid
        if guess > item_to_find:
            high = mid
        else:
            low = mid
    return None


array = [1, 4, 9, 12, 31, 47, 55, 61, 102, 204, 515]
item_to_find = int(input())

print(search(array, item_to_find))

File: 01. Binary search/Main.java
public class Main {
    public static void main(String[] args) {
        int[] array = new int[] { 1, 3, 11, 18, 39, 45, 61, 127, 193, 247, 298, 1000 };
        
        System.out.println(binarySearch(array, 18));
        System.out.println(binarySearch(array, 247));
        System.out.println(binarySearch(array, 0));
    }

    public static int binarySearch(int[] arr, int itemToFind) {
        int low = 0;
        int high = arr.length;
        int mid;

        while (low < high) {
            mid = (low + high) / 2;
            if (arr[mid] == itemToFind) {
                return mid;
            }
            if (arr[mid] < itemToFind) {
                low = mid;
            } else {
                high = mid;
            }
        }
        return -1;
    }
}